### PR TITLE
[CU-1y23865] [CU-1y23865] [CU-1yykqgq] Add a playbook to run polkadot-launch and integration tests

### PR DESCRIPTION
**Pull request by andor0**
ps: I will use build artifacts of composable, polkadot, and basilisk in the next PR.

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-1y23865 Create a playbook to run polkadot-launch](https://app.clickup.com/t/1y23865)

File: .maintain/playbooks/run-integration-tests.yml [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    get_url:
+      url: https://sh.rustup.rs
+      dest: "/home/{{ user }}/rustup-init.sh"
+      mode: '0440'
+    become: yes
+    become_user: "{{ user }}"
+
+  - name: Install Rust toolchain
+    ansible.builtin.shell: |
+      cat $HOME/rustup-init.sh | sh -s -- -y
```
Comment (dzmitry-lahoda) [inline .maintain/playbooks/run-integration-tests.yml:43]: can this be call to some SH script or ansible shared step kind of? so we avoid duplicating it a lot?

File: .maintain/playbooks/run-integration-tests.yml [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+      executable: /bin/bash
+    become: yes
+    become_user: "{{ user }}"
+
+  - name: Compile Basilisk-node
+    ansible.builtin.shell: |
+      source $HOME/.cargo/env
+      cargo build --release
+    args:
+      chdir: "{{ work_dir }}/Basilisk-node"
```
Comment (dzmitry-lahoda) [inline .maintain/playbooks/run-integration-tests.yml:176]: they have specific deps on rust, would not that require more rust intall https://github.com/galacticcouncil/Basilisk-node/blob/master/rust-toolchain ? or it is automatic?

**Comment by dzmitry-lahoda** [comment]
it may take long time to run. is any cache possible? where to see results of run? what triggers it? is fail of this script blocking?

**Review by dzmitry-lahoda — APPROVED** [review]
assuming that it will not block merges to main, but only alert. better to document that

File: .maintain/playbooks/run-integration-tests.yml (hunk 2 above)
Comment (andor0) [reply .maintain/playbooks/run-integration-tests.yml:176]: yes, nightly-2021-11-07 was installed automatically 
```
$ rustup toolchain list
stable-x86_64-unknown-linux-gnu (default)
nightly-2021-08-22-x86_64-unknown-linux-gnu
nightly-x86_64-unknown-linux-gnu
```

**Comment by andor0** [comment]
> it may take long time to run. is any cache possible?

Yes, the first run will long because need to compile composable, polkadot and basilisk, then will faster because will used an incremental compilation of cargo. I added sccache to do the first run faster.

> where to see results of run? what triggers it? is fail of this script blocking?

I will add a TLS termination for WebSocket endpoints (https://app.clickup.com/t/1y23865), then we can use wallet (https://polkadot.js.org/apps/) to connect to it. Then can add task in CI to run this playbook after a merge to main.

File: .maintain/playbooks/templates/nginx/composable-sandbox.conf.j2 [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+server {
+    listen 9901 ssl http2;
+    server_name {{ domain }};
+
+    ssl_certificate /etc/letsencrypt/live/{{ domain }}/fullchain.pem;
+    ssl_certificate_key /etc/letsencrypt/live/{{ domain }}/privkey.pem;
+
+    # Various TLS hardening settings
+    # https://raymii.org/s/tutorials/Strong_SSL_Security_On_nginx.html
```
Comment (dzmitry-lahoda) [inline .maintain/playbooks/templates/nginx/composable-sandbox.conf.j2:20]: not sure if we need so secured nginx for tests, but ok.
why do we need ngingx? to have tls rermination? please comment on that in playbook of test

Comment (dzmitry-lahoda) [reply .maintain/playbooks/templates/nginx/composable-sandbox.conf.j2:20]: so arhitecture (layers, depoyment) of test would be clear to QA and other devs

File: .maintain/playbooks/templates/nginx/composable-sandbox.conf.j2 [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    ssl_prefer_server_ciphers on;
+    ssl_ciphers 'ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256';
+    ssl_session_timeout  10m;
+    ssl_session_cache shared:SSL:10m;
+    ssl_session_tickets off;
+    ssl_stapling on;
+    ssl_stapling_verify on;
+
+    # Hide nginx version
+    server_tokens off;
```
Comment (dzmitry-lahoda) [inline .maintain/playbooks/templates/nginx/composable-sandbox.conf.j2:31]: toooo secure:)

File: .maintain/playbooks/templates/systemd/composable-sandbox.service.j2 [hunk 5]
```diff
@@ -0,0 +1,17 @@
+[Unit]
+Description=composable-sandbox
```
Comment (dzmitry-lahoda) [inline .maintain/playbooks/templates/systemd/composable-sandbox.service.j2:2]: put some meaningful description here

**Comment by andor0** [comment]
> it may take long time to run. is any cache possible? where to see results of run? what triggers it? is fail of this script blocking?

Added `sccache`.

File: .maintain/playbooks/templates/systemd/composable-sandbox.service.j2 (hunk 5 above)
Comment (andor0) [reply .maintain/playbooks/templates/systemd/composable-sandbox.service.j2]: Done.

File: .maintain/playbooks/templates/nginx/composable-sandbox.conf.j2 (hunk 3 above)
Comment (andor0) [reply .maintain/playbooks/templates/nginx/composable-sandbox.conf.j2:20]: Added README.md with details.

File: .maintain/playbooks/templates/nginx/composable-sandbox.conf.j2 (hunk 4 above)
Comment (andor0) [reply .maintain/playbooks/templates/nginx/composable-sandbox.conf.j2:31]: :)

File: .maintain/playbooks/run-integration-tests.yml (hunk 1 above)
Comment (andor0) [reply .maintain/playbooks/run-integration-tests.yml]: This isn't relevant already

**Review by dzmitry-lahoda — DISMISSED** [review]

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by dzmitry-lahoda — APPROVED** [review]

**Pull request merged**